src/scale.c

```diff
@@ -106,22 +108,15 @@ avifBool avifImageScale(avifImage * image,
             return AVIF_FALSE;
         }
 
-        avifPixelFormatInfo formatInfo;
-        avifGetPixelFormatInfo(image->yuvFormat, &formatInfo);
-        const uint32_t srcUVWidth = (srcWidth + formatInfo.chromaShiftX) >> formatInfo.chromaShiftX;
-        const uint32_t srcUVHeight = (srcHeight + formatInfo.chromaShiftY) >> formatInfo.chromaShiftY;
-        const uint32_t dstUVWidth = (dstWidth + formatInfo.chromaShiftX) >> formatInfo.chromaShiftX;
-        const uint32_t dstUVHeight = (dstHeight + formatInfo.chromaShiftY) >> formatInfo.chromaShiftY;
-
         for (int i = 0; i < AVIF_PLANE_COUNT_YUV; ++i) {
             if (!srcYUVPlanes[i]) {
                 continue;
             }
 
             const uint32_t srcW = (i == AVIF_CHAN_Y) ? srcWidth : srcUVWidth;
             const uint32_t srcH = (i == AVIF_CHAN_Y) ? srcHeight : srcUVHeight;
-            const uint32_t dstW = (i == AVIF_CHAN_Y) ? dstWidth : dstUVWidth;
-            const uint32_t dstH = (i == AVIF_CHAN_Y) ? dstHeight : dstUVHeight;
+            const uint32_t dstW = avifImagePlaneWidth(image, AVIF_CHAN_U);
+            const uint32_t dstH = avifImagePlaneHeight(image, AVIF_CHAN_U);
```

Comment:
IMPORTANT: Should we pass `i` instead of `AVIF_CHAN_U` as the second argument in lines 118-119?